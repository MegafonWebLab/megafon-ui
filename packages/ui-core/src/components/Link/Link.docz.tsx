import * as React from 'react';

export const LinkWrapper: React.FC = ({ children }) => (
    <div style={{ padding: 16 }}>{children}</div>
);
