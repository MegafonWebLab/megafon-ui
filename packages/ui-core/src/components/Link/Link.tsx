import * as React from 'react';
import * as PropTypes from 'prop-types';

interface ILinkProps {
    /** Link */
    href?: string;
    children?: JSX.Element[] | Element[] | JSX.Element | string | Element;
    /** target - property tag <a> */
    target?: '_self' | '_blank' | '_parent' | '_top';
    /** Custom class name */
    className?: string;
    /** Click event handler */
    onClick?(e: React.SyntheticEvent<EventTarget>): void;
}

class Link extends React.Component<ILinkProps, {}> {
    static propTypes = {
        href: PropTypes.string,
        children: PropTypes.oneOfType([
            PropTypes.arrayOf(PropTypes.element),
            PropTypes.element,
            PropTypes.string,
            PropTypes.node,
        ]),
        target: PropTypes.oneOf(['_self', '_blank', '_parent', '_top']),
        className: PropTypes.string,
        onClick: PropTypes.func,
    };

    render() {
        const { target, children, ...rest } = this.props;

        return (
            <a target={target} {...rest}>
                {children}
            </a>
        );
    }
}

export default Link;
