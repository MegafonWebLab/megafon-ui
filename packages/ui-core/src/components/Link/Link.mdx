---
name: Link
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Link from './Link';

# Link

## Properties

<PropsTable of={Link} />

## Basic usage

<Playground>
  <Link href="https://google.com">Click me</Link>
</Playground>
