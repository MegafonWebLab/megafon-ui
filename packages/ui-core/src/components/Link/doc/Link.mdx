---
name: Link
menu: Компоненты
route: components/link
---

import { Playground, Props } from 'docz';
import Link from '../Link';
import Example from './Link.example.mdx';

# Link

<CopyToClipboardBox text="import { Link } from '@megafon/ui-core';" />

<DoczTabs>
    <Example />
    <Props of={Link} />
</DoczTabs>
