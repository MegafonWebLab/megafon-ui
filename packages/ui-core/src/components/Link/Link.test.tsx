import * as React from 'react';
import { shallow } from 'enzyme';
import Link from './Link';

test('Link renders with default props', () => {
    const link = shallow(<Link/>);
    expect(link).toMatchSnapshot();
});

test('Link renders children', () => {
    const link = shallow(<Link><h3>some text</h3></Link>);
    expect(link).toMatchSnapshot();
});

test('Link calls onClick handler', () => {
    const fn = jest.fn();
    const link = shallow(<Link onClick={fn}>some text</Link>);

    link.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
});
