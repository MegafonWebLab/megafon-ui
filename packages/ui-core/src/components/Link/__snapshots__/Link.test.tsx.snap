// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders children 1`] = `
<a>
  <h3>
    some text
  </h3>
</a>
`;

exports[`Link renders with default props 1`] = `<a />`;
